tagless-final/src/main/java/com/github/jorgecastillo/kotlinandroid/view/SuperHeroDetailActivity.kt

```diff
@@ -40,29 +42,30 @@ class SuperHeroDetailActivity : AppCompatActivity(), SuperHeroDetailView {
     super.onResume()
     intent.extras?.let {
       val heroId = it.getString(EXTRA_HERO_ID)
-      getSuperHeroDetails(heroId, AsyncResult<GetHeroDetailsContext>()).ev().run(GetHeroDetailsContext(this, this))
+      getSuperHeroDetails(heroId, AsyncResult<GetHeroDetailsContext>()).ev()
+          .run(GetHeroDetailsContext(this, this)).value().ev().unsafeRunAsync {}
```

Comment:
You can probably encapsulate all this in an extension for `AsyncResult<D, A>.unsafePerformEffects() = ...`.
I'd expect many curious Android devs frequently opening the `Activity` files first and wondering what this chain of method calls is when they still don't have enough context to understand this is the single point where effects run.